### PR TITLE
Recipe App

### PR DESCRIPTION
**Pull request by 2004-okari**
Pull Request Description:

## Overview

This pull request enhances the user experience and functionality of the Recipe App. The primary focus is on implementing the wireframes provided while incorporating personalized design elements, such as colors, typography, and spacings. The following features have been implemented:

## Login Page and Registration Page

- Implemented using Devise for secure authentication.
- Offers a seamless and user-friendly login and registration experience.

## Recipes List

- Displays a list of recipes created by the logged-in user.
- Each recipe entry leads to its respective details page.

## Public Recipe List

- Displays a list of all public recipes, ordered by newest.
- Allows the logged-in user to delete their own recipes.
- Each recipe entry leads to its respective details page.

## Recipe Details

- Introduces a toggle button for setting a recipe as public or private.
- Displays detailed information about the recipe, considering privacy settings.
- If the user is the owner, provides access to a form for adding new food items.
- Addresses N+1 queries to optimize performance.

## Navigation Menu

- Implemented a navigation menu for easy access to all relevant pages.

## Tests

- Comprehensive unit and integration tests have been written to ensure the reliability and functionality of the implemented features.

#### Food List

- Displays a list of food added by the logged-in user, including the quantity of each item.
- Leads to a form enabling users to add new food items.

### General Shopping List View

- Shows the list of missing food items for all recipes of the logged-in user.
- Calculates the total number of food items and the overall cost of the missing items.

## Conclusion

This pull request enhances the Recipe App with a polished and intuitive interface, improved user navigation, and additional features for groups with less than 3 members. The implementation adheres to the wireframes provided and focuses on a seamless user experience. All changes have been thoroughly tested to ensure stability and functionality.

### CODE REVIEW

**Review by V-Blaze — CHANGES_REQUESTED** [review]
## Hi @2004-okari, @MargaretTechworld  

# Almost there 🟡 

giphy


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing the project requirements, and some changes from the previous reviewers, but there are still some issues I think you need to improve on. 😄 👋 


 ## Highlights
- App looks like the given wirefram✔
- Navigation and all the required pages and routes are present :heavy_check_mark: 
- All written tests are passing :heavy_check_mark: 
- Correct Git flow :heavy_check_mark: 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/recipes/public.html.erb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<div class="recipe">
+  <h3>public recipes</h3>
+  <% @recipes.each do |recipe| %>
+    <div class='recipe_box'>
+      <div class='recipe_name'>
+       <h3><%= link_to "#{recipe.name}", recipe_path(recipe.id) %></h3>
+        <p class="recipe_user">by: <%= recipe.name %></p>
```
Comment (KanzaTahreem) [inline app/views/recipes/public.html.erb:7]: - [ ] Great job so far 👏👏 However, as the previous reviewer mentioned [here](https://github.com/MargaretTechworld/recipe-app/pull/22#discussion_r1397527727) kindly make sure that the name of the recipe's owner is displaying. Currently, we have the name of the recipe in place of recipe owner's name 🐛 Kindly have a look at the screenshot attached below 📌 You can try `<%= recipe.user.inspect %>` to inspect the issue we have⛏🤓



image

**Review by KanzaTahreem — CHANGES_REQUESTED** [review]
# Changes Required 🔁

Hi @MargaretTechworld & @2004-okari 

Great work on making the changes requested by a previous reviewer 👏! I really appreciate the effort you put in!  **_However, there are few things that still need to be fixed. But you are almost there!_** 💪 🏁

so close


## Highlights 🎯

There are some excellent points that you have made perfectly. Keep up the great work! 👏
- ✅ Commented code is removed
- ✅ Great job with the design
- ✅ All implemented tests are passing

## Required Changes 🛑

_Please check the comments under the review._

## Optional suggestions ❕

N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi @MargaretTechworld  and @2004-okari , 
# STATUS: APPROVED! 🥇 💯 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**